headers.ts

```diff
@@ -0,0 +1,105 @@
+type LooseObject = object & {
```

Comment:
What's the meaning of `object & `?

Reply:
Usually you assign a value to a key in an object like this:
```typescript
const obj = {
    key: value
}
```

But if we work with headers, we want a String as key as well. Like this:
```typescript
const obj = {
    'key': value
}
```

This however gives us an error in typescript. So we need to create a custom type, the `LooseObject`. But it has to inherit everything from a normal object, that is why you see the `object &`.

With this we can use strings as keys for objects:
```typescript
const obj: LooseObject = {
    'key': value
}
```

And the AxiosHeaders type implements the LooseObject

Reply:
Sorry, I didn't get it. This does not give me any error:

```typescript
type LooseObject = {
    [key: string]: string;
};

type AxiosHeaders = LooseObject & {
    'Accept'?: string;
};

const y: AxiosHeaders = {
    Accept: 'x-any',
    'x-forwarded-for': '8.8.8.8',
    anything: 'other'
};
```

What's the case where the result with `object &` would be different?

Reply:
Yes, you are correct, my mistake. Fixed it now.